src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -384,6 +383,19 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
         return actions
     }
 
+    /**
+     * Purge all expired unpaid normal payments with creation times in the given time range.
+     *
+     * @param fromCreatedAt from absolute time in milliseconds since UNIX epoch when the payment request was generated.
+     * @param toCreatedAt to absolute time in milliseconds since UNIX epoch when the payment request was generated.
+     * @return number of invoices purged
+     */
+    suspend fun purgeExpiredPayments(fromCreatedAt: Long = 0, toCreatedAt: Long = currentTimestampMillis()): Int {
+        return db.listExpiredPayments(fromCreatedAt, toCreatedAt).map {
```

Comment:
I think you want to use `filter` instead of `map` to count only the ones that actually get removed?

Reply:
Doh! yes, actually it seems `count` applied to `removeIncomingPayment` as a predicate is what I really wanted here. Fixed in [290d8df](https://github.com/ACINQ/lightning-kmp/pull/322/commits/290d8dfcd3afb500aee2bbea91e06a14a816ca91).